tests/test_methods_and_attributes.cpp

```diff
@@ -399,4 +420,15 @@ test_initializer methods_and_attributes([](py::module &m) {
 
     using Adapted = decltype(py::method_adaptor<RegisteredDerived>(&RegisteredDerived::do_nothing));
     static_assert(std::is_same<Adapted, void (RegisteredDerived::*)() const>::value, "");
+
+    // test_custom_caster_destruction
+    // Test that `take_ownership` works on types with a custom type caster when given a pointer
+    auto dt = new DestructionTester();
```

Comment:
I want to be sure of the number of live instances of the object in the associated python tests so as to be sure there is no extra copy or extra destruction; `custom_caster_no_destroy` takes a reference (without creating anything new) while `destruction_tester_finish` is the counterpart meant to destroy that object.